lib/connection.js

```diff
@@ -91,18 +91,10 @@ Connection.prototype.connect = function (port, host) {
         return self.emit('error', new Error('There was an error establishing an SSL connection'))
     }
     var tls = require('tls')
-    const options = {
+    const options = Object.assign({
       socket: self.stream,
-      checkServerIdentity: self.ssl.checkServerIdentity || tls.checkServerIdentity,
-      rejectUnauthorized: self.ssl.rejectUnauthorized,
-      ca: self.ssl.ca,
-      pfx: self.ssl.pfx,
-      key: self.ssl.key,
-      passphrase: self.ssl.passphrase,
-      cert: self.ssl.cert,
-      secureOptions: self.ssl.secureOptions,
-      NPNProtocols: self.ssl.NPNProtocols
-    }
+      checkServerIdentity: tls.checkServerIdentity
```

Comment:
This can be taken out – `|| tls.checkServerIdentity` was only needed because it was being explicitly specified in the first place (#1638).

```suggestion
```